client/normal_nillable_array.go

```diff
@@ -84,6 +100,16 @@ func (v normalBytesNillableArray) BytesNillableArray() (immutable.Option[[][]byt
 	return v.val, true
 }
 
+func (v normalBytesNillableArray) IsEqual(other NormalValue) bool {
+	if otherVal, ok := other.BytesNillableArray(); ok {
+		if v.val.HasValue() && otherVal.HasValue() {
+			return are2DArraysEqual(v.val.Value(), otherVal.Value())
+		}
+		return !v.val.HasValue() && !otherVal.HasValue()
+	}
+	return false
```

Comment:
suggestion: Perhaps also worthwhile to hit with test if not too much hassle.

Reply:
should be covered now